src/uct/tcp/tcp.h

```diff
@@ -65,7 +65,7 @@
 #define UCT_TCP_EP_DEFAULT_KEEPALIVE_IDLE    10
 
 /* The seconds between individual keepalive probes */
-#define UCT_TCP_EP_DEFAULT_KEEPALIVE_INTVL   1
+#define UCT_TCP_EP_DEFAULT_KEEPALIVE_INTVL   10
```

Comment:
shouldn't TCP get the interval from UCP layer?

Reply:
> shouldn't TCP get the interval from UCP layer?

we should get the interval from UCP layer for `UCT_TCP_EP_DEFAULT_KEEPALIVE_IDLE` - this parameter configures interval for starting KA if TCP socket is idle

Reply:
what is the difference between them? maybe we should set both?

Reply:
after KEEPALIVE_IDLE timeout kernel starts ping peer every KEEPALIVE_INTVL seconds till get response, after this kernel waits for KEEPALIVE_IDLE again

Reply:
but I agreee that UCP should be responsible to set them. We could introduce iface input params to manage them.

Reply:
`UCT_TCP_EP_DEFAULT_KEEPALIVE_IDLE` == `UCX_KEEPALIVE_INTERVAL`, i.e. 20 second by default if KA is requested along with error handling. Otherwise, it will be `inf`, i.e. disabled. So, TCP won't send any KA packets in case of MPI/SHMEM/etc which don't require error handling.

Reply:
> after KEEPALIVE_IDLE timeout kernel starts ping peer every KEEPALIVE_INTVL seconds till get response, after this kernel waits for KEEPALIVE_IDLE again

so it seems that if we set both IDLE and INVL to same value as UCP keepalive interval, the behavior will be similar to other transports which use ep_check, where each ep_check is sent after same amount of time

Reply:
ok, I agree. Let's implement through extending `uct_iface_open()` API for built-in keepalive case?

Reply:
right, but here is issue: UCP KA may stop after N endpoints, but TCP KA will ping all EPs simultaneously.
that is why we didn't merge these values before.
not sure if this is issue, but there is different behaviour for timeouts

Reply:
> UCP KA may stop after N endpoints, but TCP KA will ping all EPs simultaneously.

I see, this can happen

Reply:
https://github.com/openucx/ucx/blob/b75809faa3e8c2015c5f5ab529ae90b0402c7ff0/src/uct/api/uct.h#L1097